crates/iroha_executor/src/default/isi/multisig/transaction.rs

```diff
@@ -0,0 +1,261 @@
+        };
+
+        if host
+            .query_single(FindAccountMetadata::new(
+                target_account.clone(),
+                approvals_key(&instructions_hash),
+            ))
+            .is_ok()
+        {
+            deny!(executor, "multisig proposal duplicates")
+        };
+    }
+
+    fn execute<V: Execute + Visit + ?Sized>(self, executor: &mut V) -> Result<(), ValidationFail> {
+        let proposer = executor.context().authority.clone();
+        let target_account = self.account;
+
+        // Authorize as the multisig account
+        executor.context_mut().authority = target_account.clone();
+
+        let instructions_hash = HashOf::new(&self.instructions);
+        let signatories: BTreeMap<AccountId, u8> = executor
+            .host()
+            .query_single(FindAccountMetadata::new(
+                target_account.clone(),
+                SIGNATORIES.parse().unwrap(),
+            ))
+            .dbg_unwrap()
+            .try_into_any()
+            .dbg_unwrap();
+        let now_ms: u64 = executor
+            .context()
+            .curr_block
+            .creation_time()
+            .as_millis()
+            .try_into()
+            .dbg_expect("shouldn't overflow within 584942417 years");
+        let approvals = BTreeSet::from([proposer]);
+
+        // Recursively deploy multisig authentication down to the personal leaf signatories
+        for signatory in signatories.keys().cloned() {
+            let is_multisig_again = executor
+                .host()
+                .query(FindRoleIds)
+                .filter_with(|role_id| role_id.eq(multisig_role_for(&signatory)))
+                .execute_single_opt()
+                .dbg_unwrap()
+                .is_some();
+
+            if is_multisig_again {
+                let propose_to_approve_me = {
+                    let approve_me =
+                        MultisigApprove::new(target_account.clone(), instructions_hash);
+
+                    MultisigPropose::new(signatory, [approve_me.into()].to_vec())
+                };
+
+                propose_to_approve_me.visit_execute(executor);
+            }
+        }
+
+        visit_seq!(executor.visit_set_account_key_value(&SetKeyValue::account(
+            target_account.clone(),
+            instructions_key(&instructions_hash).clone(),
+            Json::new(&self.instructions),
+        )));
+
+        visit_seq!(executor.visit_set_account_key_value(&SetKeyValue::account(
+            target_account.clone(),
+            proposed_at_ms_key(&instructions_hash).clone(),
+            Json::new(now_ms),
+        )));
+
+        visit_seq!(executor.visit_set_account_key_value(&SetKeyValue::account(
+            target_account,
+            approvals_key(&instructions_hash).clone(),
+            Json::new(&approvals),
+        )));
+
+        Ok(())
+    }
+}
+
+impl VisitExecute for MultisigApprove {
+    fn visit<V: Execute + Visit + ?Sized>(&self, executor: &mut V) {
```

Comment:
as far as I can see everything in this function is validated during execution

Reply:
The same as https://github.com/hyperledger-iroha/iroha/pull/5217#discussion_r1836867341